packages/interactivity/src/proxies/state.ts

```diff
@@ -0,0 +1,238 @@
+	get( target: object, key: string | symbol, receiver: object ): any {
+		/*
+		 * The property should not be reactive for the following cases:
+		 * 1. While using the `peek` function to read the property.
+		 * 2. The property exists but comes from the Object or Array prototypes.
+		 * 3. The property key is a known symbol.
+		 */
+		if (
+			peeking ||
+			( ! target.hasOwnProperty( key ) && key in target ) ||
+			( typeof key === 'symbol' && wellKnownSymbols.has( key ) )
+		) {
+			return Reflect.get( target, key, receiver );
+		}
+
+		// At this point, the property should be reactive.
+		const desc = Object.getOwnPropertyDescriptor( target, key );
+		const prop = getPropSignal( receiver, key, desc );
+
+		if ( peeking ) {
+			return prop.getComputed().peek();
+		}
+
+		const result = prop.getComputed().value;
+
+		/*
+		 * Check if the property is a synchronous function. If it is, set the
+		 * default namespace. Synchronous functions always run in the proper scope,
+		 * which is set by the Directives component.
+		 */
+		if ( typeof result === 'function' ) {
+			const ns = getProxyNs( receiver );
+			return ( ...args: unknown[] ) => {
+				setNamespace( ns );
+				try {
+					return result.call( receiver, ...args );
+				} finally {
+					resetNamespace();
+				}
+			};
+		}
+
+		return result;
+	},
+
+	set(
+		target: object,
+		key: string,
+		value: unknown,
+		receiver: object
+	): boolean {
+		setNamespace( getProxyNs( receiver ) );
+		try {
+			return Reflect.set( target, key, value, receiver );
+		} finally {
+			resetNamespace();
+		}
+	},
+
+	defineProperty(
+		target: object,
+		key: string,
+		desc: PropertyDescriptor
+	): boolean {
+		const isNew = ! ( key in target );
+		const result = Reflect.defineProperty( target, key, desc );
+
+		if ( result ) {
+			const receiver = getProxy( target );
+			const prop = getPropSignal( receiver, key );
+			const { get, value } = desc;
+			if ( get ) {
+				prop.setGetter( desc.get! );
+			} else {
+				const ns = getProxyNs( receiver );
+				prop.setValue(
+					shouldProxy( value ) ? proxifyState( ns, value ) : value
+				);
+			}
+
+			if ( isNew && objToIterable.has( target ) ) {
+				objToIterable.get( target )!.value++;
+			}
+
+			if ( Array.isArray( target ) ) {
+				const length = getPropSignal( receiver, 'length' );
+				length.setValue( target.length );
+			}
+		}
+
+		return result;
+	},
+
+	deleteProperty( target: object, key: string ): boolean {
+		const result = Reflect.deleteProperty( target, key );
+
+		if ( result ) {
+			const prop = getPropSignal( getProxy( target ), key );
+			prop.setValue( undefined );
+
+			if ( objToIterable.has( target ) ) {
+				objToIterable.get( target )!.value++;
+			}
+		}
+
+		return result;
+	},
+
+	ownKeys( target: object ): ( string | symbol )[] {
+		if ( ! objToIterable.has( target ) ) {
+			objToIterable.set( target, signal( 0 ) );
+		}
+		( objToIterable as any )._ = objToIterable.get( target )!.value;
```

Comment:
Let's add a comment, just in case someone wants to delete this line. Also, we should check in the minified bundle that it is indeed being preserved.

```suggestion
		// This subscribes to the signal while preventing the minifier from deleting this line in production. 
		( objToIterable as any )._ = objToIterable.get( target )!.value;
```